Fix window surface heat emission and radiation to air outputs

xuanluo113 opened the pull request:
Pull request overview
---------------------
- To address the issues in https://github.com/NREL/EnergyPlus/pull/7040#pullrequestreview-204070022.
- This fixes the wrong window surface heat emission calculations in the heat emission report. Previously the two variables `QAirExtReport`  and `QHeatEmiReport` are not calculated in the right place in window surfaces. 
- This also fixes output variables "Surface Outside Face Heat Emission to Air Rate" and "Surface Outside Face Thermal Radiation to Air Heat Transfer Rate" for window surfaces - before this fix these were zero

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [x] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] ~If feature, test running new feature, try creative ways to break it~
 - [ ] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] ~Check any new function arguments for performance impacts~
 - [ ] ~Verify IDF naming conventions and styles, memos and notes and defaults~
 - [ ] ~If new idf included, locally check the err file and other outputs~

xuanluo113 committed:
fix

xuanluo113 committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into fix-window-heat-emission

xuanluo113 committed:
unit test

xuanluo113 committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into fix-window-heat-emission

xuanluo113 committed:
unit test

xuanluo113 committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into fix-window-heat-emission

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into fix-window-heat-emission

mjwitte commented on src/EnergyPlus/WindowManager.cc:
```diff
@@ -3094,6 +3096,11 @@ namespace WindowManager {
+
+        // Radiation emission to air
+        QAirExtReport(SurfNum) = -SurfOutsideEmiss * sigma * (1.0 - AirSkyRadSplit(SurfNum)) * surface.ViewFactorSkyIR * (Tsout_4 - pow_4(tout));
+        QHeatEmiReport(SurfNum) = Surface(SurfNum).Area * hcout * (Tsout - tout) + QAirExtReport(SurfNum);
```
The calculation here for `QAirExtReport` duplicates a portion of the calculation above for `rad_out_per_area`, so calculate this portion separately before line 3091 and reuse it there and here.

xuanluo113 replied:
`rad_out_per_area` includes radiation from the sky, ground, and air altogether. Here `QAirExtReport` only intends to radiation to air.

mjwitte replied:
Yes, but `QAirExtReport` is part of `rad_out_per_area`. We don't want to do calculations more than necessary, especially `pow_4`.

mjwitte replied:
Here's the current code:
```
        Real64 const rad_out_per_area(
            -SurfOutsideEmiss * sigma *
            ((((1.0 - AirSkyRadSplit(SurfNum)) * surface.ViewFactorSkyIR + surface.ViewFactorGroundIR) * (Tsout_4 - pow_4(tout))) +
             (AirSkyRadSplit(SurfNum) * surface.ViewFactorSkyIR * (Tsout_4 - pow_4(SkyTempKelvin))) + QRadLWOutSrdSurfs(SurfNum)));

        QAirExtReport(SurfNum) = -SurfOutsideEmiss * sigma * (1.0 - AirSkyRadSplit(SurfNum)) * surface.ViewFactorSkyIR * (Tsout_4 - pow_4(tout));
```
This can be split apart to avoid the duplicate calculation:
```
        Real64 const rad_out_air_per_area = -SurfOutsideEmiss * sigma * (1.0 - AirSkyRadSplit(SurfNum)) * surface.ViewFactorSkyIR * (Tsout_4 - pow_4(tout));
        Real64 const rad_out_per_area = rad_out_air_per_area + (-SurfOutsideEmiss * sigma *
            ((AirSkyRadSplit(SurfNum) * surface.ViewFactorSkyIR * (Tsout_4 - pow_4(SkyTempKelvin))) + QRadLWOutSrdSurfs(SurfNum)));

        QAirExtReport(SurfNum) = surface.Area * rad_out_air_per_area;
        QHeatEmiReport(SurfNum) = surface.Area * hcout * (Tsout - tout) + QAirExtReport(SurfNum);
```
I think QAirExtReport was missing the surface area before?
Also, `QHeatEmiReport` should use `surfface.Area` instead of `Surface(SurfNum).Area`.

mjwitte commented on src/EnergyPlus/WindowManager.cc:
```diff
@@ -2463,6 +2463,8 @@ namespace WindowManager {
         using DataHeatBalSurface::QRadLWOutSrdSurfs;
         using DataHeatBalSurface::QRadOutReport;
+        using DataHeatBalSurface::QAirExtReport;
+        using DataHeatBalSurface::QHeatEmiReport;
```
No new `using` statements, just use the full name below

mjwitte reviewed (changes requested):
Tested with example file, ReportHeatEmission_RefFSRestaurant, and there are now values for "Surface Outside Face Heat Emission to Air Rate" and "Surface Outside Face Thermal Radiation to Air Heat Transfer Rate" for the window surfaces - before this fix these were zero. The numbers look reasonable relative to air and surface temperatures.

Running units tests.

xuanluo113 committed:
another fix

xuanluo113 committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into fix-window-heat-emission

xuanluo113 committed:
fix using statements

xuanluo113 commented:
@mjwitte I addressed the comments. I added another fix to calculate `QHeatEmiReport` after `QConv`, convective heat gain, is finalized. Previously the convective part in `QHeatEmiReport` is calculated by convective coefficient, which can be set to very large values during rain conditions.

xuanluo113 committed:
Merge branch 'develop' of https://github.com/NREL/EnergyPlus into fix-window-heat-emission

xuanluo113 committed:
fix

xuanluo113 committed:
fix window

xuanluo113 commented:
@mjwitte This should be clear. The diffs are due to adding the windows surface-emission values.

mjwitte committed:
Merge branch 'fix-window-heat-emission' of https://github.com/NREL/EnergyPlus into fix-window-heat-emission

mjwitte committed:
Merge remote-tracking branch 'remotes/origin/develop' into fix-window-heat-emission

mjwitte reviewed (approved):
@xuanluo113 Thanks for all the efficiency cleanup. Pulled in develop, running unit tests locally just to be sure, and I'll let CI do one more round at least on linux then merge.

Myoldmopar committed:
Merge remote-tracking branch 'origin/develop' into fix-window-heat-emission

Myoldmopar commented:
(Pulled develop in once again here)

mjwitte commented:
CI looks good with a few expected diffs. Merging.
